ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,529 @@
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface Data : IMutableScope<((int year, int month) reportingPeriod, string reportingNode, string scenario, CurrencyType currencyType, (string filterName, object filterValue)[] dataFilter)> {",
+                "\n    IDataCube<ReportVariable> InputDataCube { get; set; }",
+                "\n",
+                "\n    IDataCube<ReportVariable> DataCube { get {",
+                "\n        if(InputDataCube is null) return Enumerable.Empty<ReportVariable>().ToDataCube();",
+                "\n        var filteredDataCube = (Identity.dataFilter is null || Identity.dataFilter.Length == 0) ? InputDataCube : InputDataCube.Filter(Identity.dataFilter); ",
+                "\n        if(Identity.scenario != \"Delta\") return filteredDataCube;",
+                "\n        var bestEstimateById = filteredDataCube.Where(x => x.Scenario == null).ToDictionary(x => x.ToIdentityString());",
+                "\n        return filteredDataCube.Select(x => x.Scenario == null ? x : x with { Value = x.Value - (bestEstimateById.TryGetValue((x with {Scenario = null}).ToIdentityString(), out var be)? be.Value : 0d) }).ToDataCube();",
+                "\n    }}",
+                "\n} "
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Report Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Best Estimate PV"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "[InitializeScope(nameof(Init))]",
+                "\npublic interface PvReport : ReportScope {",
+                "\n",
+                "\n    public IDataCube<ReportVariable> GetData() => GetScopes<LockedBestEstimate>(GetDataIdentities()).Aggregate().LockedBestEstimate + GetScopes<CurrentBestEstimate>(GetDataIdentities()).Aggregate().CurrentBestEstimate;",
+                "\n",
+                "\n    async Task<GridOptions> ToReportAsync() {",
+                "\n        await GetStorage().InitializeAsync((Year, Month), ReportingNode, Scenario, CurrencyType);",
+                "\n        var dataScope = GetScope<Data>(((Year, Month), ReportingNode, Scenario, CurrencyType, dataFilter));",
+                "\n        dataScope.InputDataCube = GetData();",
+                "\n        return await GetReportTaskAsync(dataScope.DataCube);",
+                "\n    }",
+                "\n",
+                "\n     void Init() {",
+                "\n         // BasicSliceAndDiceFormsEntity",
```

Comment:
can we control from here the filters available for this report?

Reply:
Yes, it should be controlled from here